src/containers/offer-page/specialization-block/SpecializationBlock.tsx

```diff
@@ -45,7 +48,19 @@ const SpecializationBlock = <T extends CreateOrUpdateOfferData>({
   const handleCheckboxesChange = (value: ProficiencyLevelEnum[]) => {
     handleNonInputValueChange('proficiencyLevel', value)
   }
-  const levelOptions = Object.values(ProficiencyLevelEnum)
+
+  const levelLabels = useMemo(
```

Comment:
I moved the labels to the constants. However, this map can't be initialized outside the component context as it uses the `t` function. To not reevaluate this expression every time I applied `useMemo` hook.

Please let me know if that's okay. @abalanovsky @byavdoshniak

Reply:
I think it can, you need to import **i18n** and use it like `i18n.t('common.levels.beginner')` in your constants

Reply:
In that case, it will take the current language and initialize the map once. If you will change the lang - the map won't be reinitialized.

Reply:
There is another option we can but it's less flexible. We can pass the map of codes to the `CheckboxList` component and then the checkbox will use its `t` to translate the code. Additionally, we can introduce some kind of labels options props where we can specify if the label should be treated as a text or as a code (in that case the checkbox will use the `t` to translate the code)